src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -832,18 +840,62 @@ void *frontendservicesesion(std::string masterIP, int connFd, SQLiteDBInterface
                 string error_message = "Graph is not in the active status";
                 frontend_logger.log(error_message, "error");
                 result_wr = write(connFd, error_message.c_str(), error_message.length());
-                if(result_wr < 0) {
+                if (result_wr < 0) {
                     frontend_logger.log("Error writing to socket", "error");
                 }
                 result_wr = write(connFd, "\r\n", 2);
-                if(result_wr < 0) {
+                if (result_wr < 0) {
                     frontend_logger.log("Error writing to socket", "error");
                 }
                 continue;
             }
 
             JasminGraphTrainingInitiator *jasminGraphTrainingInitiator = new JasminGraphTrainingInitiator();
-            jasminGraphTrainingInitiator->initiateTrainingLocally(graphID,trainData);
+            jasminGraphTrainingInitiator->initiateTrainingLocally(graphID, trainData);
+        } else if (line.compare(IN_DEGREE) == 0)  {
+            frontend_logger.log("Calculating In Degree Distribution", "info");
+
+            int result_wr = write(connFd, SEND.c_str(), FRONTEND_COMMAND_LENGTH);
+            if (result_wr < 0) {
+                frontend_logger.log("Error writing to socket", "error");
+                loop = true;
+                continue;
+            }
+            result_wr = write(connFd, "\r\n", 2);
+            if (result_wr < 0) {
+                frontend_logger.log("Error writing to socket", "error");
+                loop = true;
+                continue;
+            }
+
+            char graph_id[FRONTEND_DATA_LENGTH];
+            bzero(graph_id, FRONTEND_DATA_LENGTH + 1);
+            string name = "";
```

Comment:
Do we ever use `name` and `path` variables here?

Reply:
Removed